Hw03 spring boot

Greyzl opened the pull request.

Commit:
1.  Init project

Commit:
1.  Init project

Commit:
Добавил зависимости через стартеры. Перевел мейн на @SpringBootApplication

Commit:
Добавил зависимости через стартеры. Перевел мейн на @SpringBootApplication

Commit:
Добавил локализацию

Commit:
Добавил локализацию

Commit:
Добавил инъекцию в тесто по Dao

Commit:
Добавил инъекцию в тесто по Dao

Commit:
Поправил тесты

Commit:
Поправил тесты

Commit:
Merge remote-tracking branch 'origin/hw03-spring_boot' into hw03-spring_boot

Commit:
Добавил баннер

stvort commented on hw03-spring_boot/src/main/java/ru/otus/dao/impl/QuestionCSVDao.java:
```diff
@@ -0,0 +1,49 @@
+
+    private final String fileName ;
+
+    public QuestionCSVDao(QuestionMapperCsv questionMapperCsv, AppProps props){
```
Данному классу не нужны все настройки приложения, только имя файла. Решение проблемы мы обсуждали на разборе ДЗ и на занятии по архитектуре

stvort commented on hw03-spring_boot/src/main/java/ru/otus/formatter/impl/TestResultOutputFormatterImpl.java:
```diff
@@ -0,0 +1,29 @@
+
+    private final MessageSource messageSource;
+
+    private final AppProps appProps;
```
Эти вещи лучше скрыть за сервисом, чтобы сократить количество внедряемых зависимостей и упросить получение перевода по ключу. Таком сервису тоже не будут нужны все настройки

stvort commented on hw03-spring_boot/src/test/java/ru/otus/dao/impl/QuestionCSVDaoTest.java:
```diff
@@ -0,0 +1,41 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+
+@SpringBootTest
+class QuestionCSVDaoTest {
```
Переход на `@SpringBootTest` должен был случиться только в следующей работе. Теперь пусть остается)

Commit:
Создал сервис по работе с локализируемыми сообщениями
Добавил два интерфейса для работы с пропертями из application.yml и внедрил в DAO и LocalizedMessageService

stvort commented on hw03-spring_boot/src/main/java/ru/otus/service/impl/LocalizedMessageServiceImpl.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    @Override
+    public String getMessage(String key, Object[] args) {
```
Если сделать вараргами, будет короче вызыов
